app/models/exchange_variant.rb

```diff
@@ -3,9 +3,13 @@
 class ExchangeVariant < ApplicationRecord
   belongs_to :exchange
   belongs_to :variant, class_name: 'Spree::Variant'
-  after_destroy :destroy_related_outgoing_variants
 
-  def destroy_related_outgoing_variants
-    VariantDeleter.new.destroy_related_outgoing_variants(variant_id, exchange.order_cycle)
+  after_destroy :delete_related_outgoing_variants
+
+  def delete_related_outgoing_variants
+    ExchangeVariant.where(variant_id: variant_id).
+      joins(:exchange).
+      where(exchanges: { order_cycle: exchange.order_cycle, incoming: false }).
+      delete_all
```

Comment:
This seems correct when destroying an incoming variant. But what about outgoing variants? If you have an OC with several distributors and two distributors sell the same variant then removing the variant from one shop would also remove it from the other, right?

That seems to be a pre-existing though. My mistake for not seeing this earlier because I reviewed it.

So if I'm right, we could probably just add `return unless exchange.incoming?` or add the condition to the callback.